docs/content/docs/custom-resource/reference.md

```diff
@@ -174,6 +174,79 @@ This page serves as a full reference for FlinkDeployment custom resource definit
 | LAST_STATE | Job is upgraded using any latest checkpoint or savepoint available. |
 | STATELESS | Job is upgraded with empty state. |
 
+### Diff
+**Class**: org.apache.flink.kubernetes.operator.crd.spec.diff.Diff
+
+**Description**: Contains the differences between two {@link Diffable} class fields.
+
+ <p>Inspired by:
+ https://github.com/apache/commons-lang/blob/master/src/main/java/org/apache/commons/lang3/builder/Diff.java
+
+| Parameter | Type | Docs |
+| ----------| ---- | ---- |
+| serialVersionUID | long |  |
+| fieldName | @lombok.NonNull java.lang.String |  |
+| left | T |  |
+| right | T |  |
+| type | @lombok.NonNull org.apache.flink.kubernetes.operator.crd.spec.diff.DiffType |  |
+
```

Comment:
These Diff related classes should not be part of the CRD reference. We should either move them to another package or change the generator logic.